lockable_slice.go

```diff
@@ -78,6 +78,7 @@ func (this *_LockableSlice) _Advance() int { //nolint
 	index := this.index
 	if len(this.slice) != 0 {
 		this.index = (this.index + 1) % len(this.slice)
+		index = this.index
```

Comment:
Not sure if this is what we want to do. You're free to deviate from what we have In JS, but initially what I had in mind for `Advance` was to increment index and return previous index, not return the new incremented index. This change feels like it should cause some tests to fail and a bit concerned if that wasn't the case.

Reply:
It seems nothing actually uses the returned int.